Block/Adminhtml/System/Config/Fieldset/Payment.php

```diff
@@ -0,0 +1,199 @@
+use Magento\Framework\Module\Dir\Reader;
+
+class Payment extends \Magento\Paypal\Block\Adminhtml\System\Config\Fieldset\Payment
+{
+
+    const MODULE_CODE = 'Amazon_Pay';
+    const MODULE_PACKAGE_NAME = 'amzn/amazon-pay-magento-2-module';
+    const MODULE_COMPOSER_FILE = '/composer.json';
+    const MODULE_CHANGELOG_URL = 'https://github.com/amzn/amazon-payments-magento-2-plugin/blob/master/CHANGELOG.md';
+    /**
+     * @var SecureHtmlRenderer
+     */
+    private $secureRenderer;
+
+    /**
+     * @var Reader
+     */
+    private $moduleReader;
+
+    /**
+     * @var File
+     */
+    private $filesystem;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    public function __construct(
+        Context $context,
+        Session $authSession,
+        Js $jsHelper,
+        Config $backendConfig,
+        Reader $moduleReader,
+        File $filesystem,
+        Serializer $serializer,
+        Client $client,
+        SecureHtmlRenderer $secureHtmlRenderer,
+        array $data = [],
+
+    ) {
+        parent::__construct($context, $authSession, $jsHelper, $backendConfig, $data, $secureHtmlRenderer);
+
+        $this->secureRenderer = $secureHtmlRenderer;
+        $this->moduleReader = $moduleReader;
+        $this->filesystem = $filesystem;
+        $this->serializer = $serializer;
+        $this->client = $client;
+    }
+
+    /**
+     * Read version about extension from composer json file
+     *
+     * @param string $moduleCode
+     *
+     * @return mixed
+     */
+    public function getLocalModuleVersion(): mixed
+    {
+        try {
+            $dir = $this->moduleReader->getModuleDir('', self::MODULE_CODE);
+            $file = $dir . self::MODULE_COMPOSER_FILE;
+
+            $jsonFileContent = $this->filesystem->fileGetContents($file);
+            $fileData = $this->serializer->unserialize($jsonFileContent);
+            $version = $fileData['version'];
+
+        } catch (FileSystemException $e) {
+            $version = false;
+        }
+
+        return $version;
+    }
+
+    /**
+     * @return mixed|string
+     */
+    public function getRemoteModuleVersion(): mixed
+    {
+        $lastVersion = '';
+
+        try {
+            $availableVersionsData = json_decode($this->client->getJsonData(), true);
+            $lastVersionData = $availableVersionsData['packages'][self::MODULE_PACKAGE_NAME][0];
+            $lastVersion = $lastVersionData['version'];
+        } catch (\Exception $e) {
+            //nothing
+        }
+
+        return $lastVersion;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isLastVersion() :bool
+    {
+        return version_compare(
+            $this->getLocalModuleVersion(),
+            $this->getRemoteModuleVersion(),
+            '<='
+        );
+    }
+
+    /**
+     * @return string
+     */
+    protected function renderUpdateNotification(): string
```

Comment:
An improvement could be here made by separating the HTML rendering to a separate **block class/view model** with .phtml template.
`$updateNotificationBlock = $this->layout->createBlock(<block_path>);
        return $updateNotificationBlock->toHtml();`
where the template path is set in the constructor or as an overridden property in the block class.